### ISSUE TITLE
ESLint server only looks at first entry in `context.only` array sent along with `textDocument/codeAction`

### ISSUE THREAD

**Issue by mrnugget** [microsoft/vscode-eslint#1905]
At Zed we allow users to set up multiple code actions to be executed on save.

That breaks with the ESLint language server because the server only looks at the first entry of the `context: { only: [] }` array sent along with `textDocument/codeAction` requests.

Example: when we sent a request like this

```json
{
  "jsonrpc": "2.0",
  "id": 140,
  "method": "textDocument/codeAction",
  "params": {
    // ...
    "context": {
      "diagnostics": [
        // ...
      ],
      "only": ["source.organizeImports", "source.fixAll.eslint"]
    }
  }
}
```

Then the ESLint server will only look at `"source.organizeImports"` and ignore the `source.fixAll.eslint`. And that has disastrous results: the server will add "disable" comments when executing `organizeImports` — which is not what users want most of the time.

@hmttrp tracked the behavior down to the code here:

https://github.com/microsoft/vscode-eslint/blob/edc9685bcda26462dc52decce213167af8d2b25d/server/src/eslintServer.ts#L588

I'm not sure what exactly the intention behind the check is (I don't understand why not all entries are checked), but I think a fix would be to turn the booleans here

https://github.com/microsoft/vscode-eslint/blob/edc9685bcda26462dc52decce213167af8d2b25d/server/src/eslintServer.ts#L589-L590

into checks whether the array contains an item of that kind (and not just whether the first item has that kind).

**Comment by dbaeumer** [comment]
This is a problem with the Zed client. The ESLint server registers for the following code action kinds

https://github.com/microsoft/vscode-eslint/blob/main/server/src/eslintServer.ts#L224

It should not be called with a code action kind of organize imports since this is nothing it has been registered for.

**Comment by mrnugget** [comment]
That's how I ended up fixing it: https://github.com/zed-industries/zed/pull/14666

Ultimately your call, of course, but I think being more robust against input here and not executing different code actions would make future debugging easier.

**Comment by dbaeumer** [comment]
I agree with that. But a client should not send unnecessary code actions requests for performance reasons.

**Comment by mrnugget** [comment]
Yeah, 100%! That was a bug on our side, definitely!
